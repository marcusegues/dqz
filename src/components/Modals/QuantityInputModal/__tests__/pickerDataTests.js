// @flow

import {
  amountsPicker,
  pickerDates,
  pickerHours,
  pickerItemGenerator,
  pickerMinutes,
  rangeItemsPicker,
} from '../pickerData';

describe('test the data', () => {
  test('picker hours', () => {
    expect(pickerHours.length).toBe(24);
  });

  test('picker minutes', () => {
    expect(pickerMinutes.length).toBe(12);
  });
});

describe('picker generator', () => {
  const expected = JSON.stringify(pickerItemGenerator(['a', 'b', 'c']));
  const returned = JSON.stringify([
    { id: 0, label: 'a', value: 'a' },
    { id: 1, label: 'b', value: 'b' },
    { id: 2, label: 'c', value: 'c' },
  ]);
  expect(expected).toBe(returned);
});

describe('AmountsPicker', () => {
  test('does the right thing', () => {
    expect(amountsPicker('AlcSoft')).toMatchSnapshot();
    expect(amountsPicker('AlcHard')).toMatchSnapshot();
    expect(amountsPicker('Cigarettes')).toMatchSnapshot();
    // $FlowFixMe
    expect(amountsPicker('Foobar')).toMatchSnapshot();
  });
});

describe('rangeItemsPicker', () => {
  test('does the right thing', () => {
    const expected = JSON.stringify(rangeItemsPicker(0, 3, 1));
    const returned = JSON.stringify([
      { id: 0, label: '0', value: '0' },
      { id: 1, label: '1', value: '1' },
      { id: 2, label: '2', value: '2' },
      { id: 3, label: '3', value: '3' },
    ]);
    expect(expected).toBe(returned);
  });
  test('does the right thing with larger steps', () => {
    const expected = JSON.stringify(rangeItemsPicker(0, 3, 2));
    const returned = JSON.stringify([
      { id: 0, label: '0', value: '0' },
      { id: 1, label: '2', value: '2' },
    ]);
    expect(expected).toBe(returned);
  });
});

describe('pickerHours', () => {
  expect(pickerDates.length).toBe(3);
  expect(pickerDates.map(p => p.id).toString()).toBe([0, 1, 2].toString());
});
