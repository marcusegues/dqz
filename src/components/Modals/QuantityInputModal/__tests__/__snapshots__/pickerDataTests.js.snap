// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountsPicker does the right thing 1`] = `
Array [
  Object {
    "id": 0,
    "label": "0.33",
    "value": "0.33",
  },
  Object {
    "id": 1,
    "label": "0.5",
    "value": "0.5",
  },
  Object {
    "id": 2,
    "label": "0.75",
    "value": "0.75",
  },
  Object {
    "id": 3,
    "label": "1",
    "value": "1",
  },
  Object {
    "id": 4,
    "label": "1.5",
    "value": "1.5",
  },
  Object {
    "id": 5,
    "label": "2",
    "value": "2",
  },
]
`;

exports[`AmountsPicker does the right thing 2`] = `
Array [
  Object {
    "id": 0,
    "label": "0.33",
    "value": "0.33",
  },
  Object {
    "id": 1,
    "label": "0.5",
    "value": "0.5",
  },
  Object {
    "id": 2,
    "label": "0.75",
    "value": "0.75",
  },
  Object {
    "id": 3,
    "label": "1",
    "value": "1",
  },
  Object {
    "id": 4,
    "label": "1.5",
    "value": "1.5",
  },
  Object {
    "id": 5,
    "label": "2",
    "value": "2",
  },
]
`;

exports[`AmountsPicker does the right thing 3`] = `
Array [
  Object {
    "id": 0,
    "label": "1",
    "value": "1",
  },
  Object {
    "id": 1,
    "label": "10",
    "value": "10",
  },
  Object {
    "id": 2,
    "label": "20",
    "value": "20",
  },
  Object {
    "id": 3,
    "label": "50",
    "value": "50",
  },
  Object {
    "id": 4,
    "label": "100",
    "value": "100",
  },
  Object {
    "id": 5,
    "label": "200",
    "value": "200",
  },
  Object {
    "id": 6,
    "label": "250",
    "value": "250",
  },
  Object {
    "id": 7,
    "label": "500",
    "value": "500",
  },
]
`;

exports[`AmountsPicker does the right thing 4`] = `
Array [
  Object {
    "id": 0,
    "label": "1",
    "value": "1",
  },
  Object {
    "id": 1,
    "label": "2",
    "value": "2",
  },
  Object {
    "id": 2,
    "label": "5",
    "value": "5",
  },
  Object {
    "id": 3,
    "label": "10",
    "value": "10",
  },
  Object {
    "id": 4,
    "label": "50",
    "value": "50",
  },
  Object {
    "id": 5,
    "label": "100",
    "value": "100",
  },
  Object {
    "id": 6,
    "label": "200",
    "value": "200",
  },
  Object {
    "id": 7,
    "label": "1000",
    "value": "1000",
  },
]
`;
